python/lsst/ts/Dome/mock_llc_statuses/amcs_status.py

```diff
@@ -55,23 +54,34 @@ async def determine_status(self, current_tai):
         )
         if self.status != LlcStatus.STOPPED.value:
             azimuth_step = self.motion_velocity * time_diff
-            if self.motion_direction == motion_dir.CW.value:
-                self.position_actual = self.position_orig + azimuth_step
+            self.position_actual = self.position_orig + azimuth_step
+            if self.motion_velocity >= 0:
                 if self.position_actual >= self.position_cmd:
                     self.position_actual = self.position_cmd
-                    self.position_orig = self.position_actual
                     if self.status == LlcStatus.MOVING.value:
-                        self.status = LlcStatus.STOPPED.value
+                        if self.crawl_velocity >= 0:
+                            self.position_cmd = math.radians(720)
+                        else:
+                            self.position_cmd = math.radians(-360)
+                        self.motion_velocity = self.crawl_velocity
+                        self.status = LlcStatus.CRAWLING.value
```

Comment:
Prior code, but please consider setting self.status to the actual enum, instead of its value (e.g. `self.status = LlcStatus.CRAWLING`.

That simplifies the code, allows you to benefit from knowing status is an enum, not an arbitrary int, and if you ever print the value you get the name, instead of an integer that is hard to understand.

Reply:
The actual value is a string, not an integer. Having said that, I agree that this makes the code easier to read so I made the change in all status classes apart from the one for the louvers since numpy arrays don't seem to appreciate string-based enums.

Reply:
On top of that, I really need to have the value of the Enum in the status string, and not the Enum itself. So even if the numpy array can hold Enums, I will need to add an additional step to get the values instead.

Reply:
I am confused about this. Why use string values for enums? Ints should work just fine. When you print the enum  you get the name.